src/state_migration/nv18/migration.rs

```diff
@@ -34,28 +35,24 @@ impl<BS: Blockstore + Clone + Send + Sync> StateMigration<BS> {
         let system_actor_state = store
             .get_cbor::<SystemStateOld>(&system_actor.state)?
             .ok_or_else(|| anyhow!("system actor state not found"))?;
-        let current_manifest_data = system_actor_state.builtin_actors;
-        let current_manifest = ManifestOld::load(&store, &current_manifest_data, 1)?;
+        let current_manifest =
+            Manifest::load_with_actors(&store, &system_actor_state.builtin_actors, 1)?;
 
-        let (version, new_manifest_data): (u32, Cid) = store
-            .get_cbor(new_manifest)?
-            .ok_or_else(|| anyhow!("new manifest not found"))?;
-        let new_manifest = ManifestNew::load(&store, &new_manifest_data, version)?;
+        let new_manifest = Manifest::load(&store, new_manifest)?;
 
-        current_manifest.builtin_actor_codes().for_each(|code| {
-            let id = current_manifest.id_by_code(code);
-            let new_code = new_manifest.code_by_id(id).unwrap();
+        current_manifest.builtin_actors().for_each(|(name, code)| {
+            let new_code = new_manifest.code_by_name(name).unwrap();
```

Comment:
Are you sure we want to panic in such a case? If so, let's at least provide a helpful message with `expect`.

Reply:
The original logic uses unwrap and the error message in code_by_name should be clear enough

Reply:
```rust
pub fn code_by_name(&self, name: &str) -> anyhow::Result<&Cid> {
        self.by_name
            .get(name)
            .ok_or_else(|| anyhow!("Failed to retrieve actor code by name {name}"))
    }
```